index.js

```diff
@@ -78,17 +79,22 @@ export class StuckTransactionsCanceller {
 
   async removeConfirmed (tx) {
     assert.strictEqual(typeof tx.hash, 'string')
-    await this.#store.remove(tx.hash)
+    const txs = await this.#store.list()
+    for (const _tx of txs) {
+      if (_tx.nonce === tx.nonce) {
+        await this.#store.remove(_tx.hash)
+      }
+    }
```

Comment:
Potential improvement:

Add a new Store method `listByNonce(nonce)` that performs the filtering at the DB server and does not need to transfer all TXs from the DB to the Node.js process.

We can also consider adding another method to batch-remove transactions (e.g. `removeByNonce([nonce1, nonce2,...])` returning a list of TX hashes that were removed).

This is not needed now that we have less than 1000 pending transactions, but we should keep this potential performance bottleneck in mind.

Reply:
I don't think anything tells us that this complexity is worth it. Even with a million transactions I don't think this matter.